### DISCUSSION TITLE
Howto / is Pinnacle ready...

### DISCUSSION THREAD

**Discussion by ildar** [pinnacle-comp/pinnacle#151] [Q&A] [answered]
to be used in a really advanced fashion?

E.g. I recently used [the-glorious-dotfiles](https://github.com/eromatiya/the-glorious-dotfiles) which is the advanced Lua configuration on top of the AwesomeWM.
**Q**: is Pinnacle ready for similar advanced Lua configuration?
Can someone share his/her eyecandy Pinnacle configuration?

**Comment by Ottatop** [comment, marked as answer]
> ...to be used in a really advanced fashion?

Pinnacle is currently in a pre-alpha state. The API is only capable of doing the bare minimum of what a compositor needs to do, and the actual feature set of the compositor is pretty limited. Also there are quite a few bugs. So no, not right now.

> **Q**: is Pinnacle ready for similar advanced Lua configuration?

Not at this moment. There's not even a widget system yet or anything resembling Awesome's signals and theming, so you're not going to be able to do much. It's pretty much just setting key- and mousebinds, setting up monitors, limited manipulation of windows, and spawning programs.

> Can someone share his/her eyecandy Pinnacle configuration?

I *doubt* anyone has tried using Pinnacle for an extended period of time (even me), and even then any eyecandy would be from external tools like waybar and eww.

Obviously in the future I'd expect all of these to get better, but that's a long time off. Plus I'm at university right now so development is going to be slow for the next few months. Maybe check back in... 3 months? 6 months? A year? I don't really know how long any of this is going to take.

**Comment by ildar** [comment]
At last I managed to test Pinnacle with Lua configs.
Yes, it is indeed very young and lacks some basic features >> Things I stumbled upon instantly
were: sloppy focus and fractional scaling. I found them in the list here: https://github.com/pinnacle-comp/pinnacle/issues/142
But one thing I couldn't do (that's not in TODO): bind a key for window switching (classical Alt+Tab behavior)

**Reply by Ottatop** [reply]
> But one thing I couldn't do (that's not in TODO): bind a key for window switching (classical Alt+Tab behavior)

Yea I still need to add some sort of `window:set_focus` method so you can do that.

Once I add that and finish working on #158 then sloppy focus should also be possible using something like
```lua
Window.connect_signal({
    pointer_enter = function(window)
        window:set_focus(true)
    end
})
```

> At last I managed to test Pinnacle with Lua configs.

Just curious, this sounds like it took a while to set up the project. Were there any difficulties in doing so? I know there are quite a few more setup steps than other compositors, so any insight on how to make setup easier for newcomers is greatly appreciated.

**Comment by ildar** [comment]
hi!

On Tue, Feb 13, 2024 at 1:53 AM Ottatop wrote:
> Yea I still need to add some sort of window:set_focus method so you can do that.
> Once I add that and finish working on #158 then sloppy focus should also be possible

that's great 👍

> Just curious, this sounds like it took a while to set up the project. Were there any difficulties in doing so? I know there are quite a few more setup steps than other compositors, so any insight on how to make setup easier for newcomers is greatly appreciated.

yes, it's mostly nothing related to Pinnacle itself. I prepared some boilerplate stuff — packaged required LuaRocks to RPM.

The things that still need to be fixed are:
1. Put default configs to some system place (e.g. /etc/pinnacle)
2. change Protobuf files location to e.g. /usr/share/pinnacle
Maybe it's ok to keep the old location (~/.loca/share/pinnacle) as the search path.

I understand you're busy with your studies. I'd suggest you concentrate on the university as this is much more important. And put a small note to the README that the project is on hold. Just IMHO

**Comment by ildar** [comment]
On Mon, Feb 12, 2024 at 11:04 AM Ildar Mulyukov wrote:
> Things I stumbled upon instantly were: sloppy focus and fractional scaling.
Just tested the sloppy focus and it works!!! Congrats!

> But one thing I couldn't do (that's not in TODO): bind a key for window switching (classical Alt+Tab behavior)

So it seems when the scaling is ready I may try to switch from
Hyprland to Pinnacle.
Thanks! And all the best to your studies!

**Comment by ildar** [comment]
So Pinnacle is actually READY and I switched to it ~1 month ago.
Thanks, @Ottatop!!!
P.S. Still not as eyecandy as e.g. Hyprland. But favoring Pinnacle anyway!
